components/tree/private/branch.jsx

```diff
@@ -69,6 +73,87 @@ const handleScroll = (event, props) => {
 	}
 };
 
+const findNextNode = (flattenedNodes, node) => {
+	const nodes = flattenedNodes.map(((flattenedNode) => flattenedNode.node));
+	const index = nodes.indexOf(node);
+	return flattenedNodes[(index + 1) % flattenedNodes.length];
+};
+
+const findPreviousNode = (flattenedNodes, node) => {
+	const nodes = flattenedNodes.map(((flattenedNode) => flattenedNode.node));
+	let index = nodes.indexOf(node) - 1;
+	if (index < 0) {
+		index += flattenedNodes.length;
+	}
+	return flattenedNodes[index];
+};
+
+const handleKeyDownDown = (event, props) => {
+	if (props.node.selected) {
+		// Go to the next visible node
+		handleClick(event, props);
+		const flattenedNode = findNextNode(props.flattenedNodes, props.node);
+		props.onClick(event, {
+			node: flattenedNode.node,
+			select: true,
+			treeIndex: flattenedNode.treeIndex,
+		});
+	} else {
+		// Focus is on this node but it's not selected, so select it
+		handleClick(event, props);
+	}
+};
+
+const handleKeyDownUp = (event, props) => {
+	if (props.node.selected) {
+		// Go to the next visible node
+		handleClick(event, props);
+		const flattenedNode = findPreviousNode(props.flattenedNodes, props.node);
+		props.onClick(event, {
+			node: flattenedNode.node,
+			select: true,
+			treeIndex: flattenedNode.treeIndex,
+		});
+	} else {
+		// Focus is on this node but it's not selected, so select it
+		handleClick(event, props);
+	}
+};
+
+const handleKeyDownRight = (event, props) => {
+	if (!props.node.expanded) {
+		handleExpandClick(event, props);
+	}
+};
+
+const handleKeyDownLeft = (event, props) => {
+	if (props.node.expanded) {
+		handleExpandClick(event, props);
+	} else {
+		const nodes = props.flattenedNodes.map(((flattenedNode) => flattenedNode.node));
+		const index = nodes.indexOf(props.parent);
+		if (index !== -1) {
+			handleClick(event, props);
+			props.onClick(event, {
+				node: props.parent,
+				select: true,
+				treeIndex: props.flattenedNodes[index].treeIndex,
+			});
+		}
+	}
+};
+
+const handleKeyDown = (event, props) => {
+	mapKeyEventCallbacks(event, {
+		callbacks: {
```

Comment:
We'll need an `enter` public callback and all the multi-select stuff per the SLDS spec here.

Reply:
I tied `Enter` to the item click callback.  Is that what you mean?

Reply:
I confirmed that click and enter should be the same callback.